scripts/cdn_deploy.js

```diff
@@ -3,124 +3,132 @@
-				const ext = relativePath.substring(extIndex);
-				const newPath = `${relativePath.substring(0, extIndex)}-${version}${ext}`;
-				let fileData = fs.readFileSync(file).toString();
-				if(newPath.endsWith(".min.js"))
-					fileData = fileData.replace("//# sourceMappingURL=ably.min.js.map", `//# sourceMappingURL=${newPath}.map`);
-				await upload(s3, {
-					Body: fileData,
-					Key: path.join(config.root, newPath),
-					Bucket: config.bucket,
-				});
-			}
-		}
-		console.log("Success!");
-	} catch (e) {
-		throw new Error(e);
-	}
+  let config = {
+    // The S3 Bucket to upload into
+    bucket: S3_DEFAULT_BUCKET,
+    // The root folder inside the S3 bucket where the files should be places
+    root: S3_DEFAULT_ROOT,
+    // Local path to start from
+    path: '.',
+    // Comma separated directories (relative to `path`) to upload
+    includeDirs: 'browser/static',
+    // Comma separated directories (relative to `path`) to exclude from upload
+    excludeDirs: 'node_modules,.git',
+    // Regex to match files against for upload
+    fileRegex: '^ably(\\.noencryption)?(\\.min)?\\.js(\\.map)?',
+    ...argv,
+  };
+
+  // Resolve all the paths into full paths
+  config.path = path.resolve(config.path);
+  config.includeDirs = config.includeDirs.split(',').map((dir) => path.resolve(dir));
+  config.excludeDirs = config.excludeDirs.split(',').map((dir) => path.resolve(dir));
+
+  const s3 = new AWS.S3();
+
+  // If no tag is specified, run an output displaying all available tags
+  if (!config.tag) {
+    let refs = await git('tag');
+    console.log('Available tags:');
+    console.log(refs);
+    throw new Error('You must supply a tag with --tag or skip this with --force');
+  }
+
+  const isTag = await git('tag --points-at HEAD');
+  if (!isTag) throw new Error(`Tag name '${config.tag}' does not point at HEAD.`);
+
+  if (!config.skipCheckout) await git(`checkout tags/${config.tag}`);
+
+  console.log(`Starting Ably Javascript S3 library deployment`);
+  console.log('Bucket:', config.bucket);
+  console.log('Output Root:', config.root);
+  console.log('Input Path:', config.path);
+  console.log('Included Dirs:', config.includeDirs.join(', '));
+  console.log('Excluded Dirs:', config.excludeDirs.join(', '));
+  console.log('File Regex:', config.fileRegex);
+  console.log('-------');
+
+  const files = recursiveFile(config.includeDirs, config.excludeDirs, config.fileRegex);
+  const versions = getVersions(config.tag);
+  console.log(`Found ${files.length} files to upload...`);
+  try {
+    for (let file of files) {
+      console.log(`Uploading ${file}...`);
+      for (let version of versions) {
+        const relativePath = path.relative(
+          config.includeDirs.find((d) => file.startsWith(d)),
+          file
+        );
+        const split = relativePath.split('.js');
+        const newPath = `${split[0]}-${version}.js${split[1]}`;
+        let fileData = fs.readFileSync(file).toString();
+        let ContentType;
```

Comment:
I'm not sure why this is cased this way - why not `contentType`? 😕

Reply:
I did that so that I can use the shorthand object property assignment on line 83 of that file:
```ts
        await upload(s3, {
          Body: fileData,
          Key: path.join(config.root, newPath),
          Bucket: config.bucket,
          ContentType,
        });
```
As for why AWS decided to name their properties like that I have no clue 🤷

Reply:
Interesting... breaking a naming convention to take advantage of a language feature. 😆 

Not sure I'm with you there, but fair enough, thanks for explaining. If you prefer this 'look' then keep it as is. 😝